parquet/src/main/java/org/apache/iceberg/parquet/Parquet.java

```diff
@@ -582,6 +582,13 @@ public ReadBuilder withNameMapping(NameMapping newNameMapping) {
           optionsBuilder = ParquetReadOptions.builder();
         }
 
+        if (filter != null &&
+                schema.getAliases() != null &&
+                ParquetFilters.isSupportedFilter(filter, schema, caseSensitive)) {
+          optionsBuilder.useRecordFilter(filterRecords);
+          optionsBuilder.withRecordFilter(ParquetFilters.convert(schema, filter, caseSensitive));
```

Comment:
Do you recommend to move "type = schemaReader.getFileMetaData().getSchema()" to the beginning of this method so that we can use the file schema instead of table schema? I worry it could result in more storage API calls because when the condition "if (readerFunc != null || batchedReaderFunc != null)" is true, we don't have this API call today.

Reply:
We don't have to move it to the absolute top, we can only fetch the file schema if the filter is supposed to be used. It will lead to an extra open call, but I am not sure I see a good way around it.

Reply:
Yes, we call it when we need to convert.